scanner/src/main/kotlin/scanners/scanoss/ScanOss.kt

```diff
@@ -47,30 +56,79 @@ class ScanOss internal constructor(
             ScanOss(scannerName, scannerConfig, downloaderConfig)
     }
 
+    private val config = ScanOssConfig.create(scannerConfig).also {
+        log.info { "The $scannerName API URL is ${it.apiUrl}." }
+    }
+
+    private val service = ScanOssService.create(config.apiUrl)
+
+    companion object {
+        // The name of the multipart body being sent. Has no importance.
+        private const val FAKE_WFP_FILE_NAME = "anon.wfp"
+    }
+
+    // TODO: Find out the best / cheapest way to query the SCANOSS server for its version.
     override val version = BuildConfig.SCANOSS_VERSION
+
     override val configuration = ""
 
+    /**
+     * The name of the file corresponding to the fingerprints can be sent to SCANOSS for more precise matches.
+     * However, for anonymity, a unique identifier should be generated and used instead. This property holds the
+     * mapping between the file paths and the unique identifiers. When receiving the response, the UUID will be
+     * replaced by the actual file path.
+     *
+     * TODO: This behaviour should be driven by a configuration parameter enabled by default.
+     */
+    private val fileNamesAnonymizationCache = mutableMapOf<UUID, String>()
+
     override fun scanPathInternal(path: File): ScanSummary {
         val resultFile = createOrtTempDir().resolve("result.json")
         val startTime = Instant.now()
 
-        // TODO: Support API configurations other than OSSKB.
-        val scannerConf = ScannerConf.defaultConf()
-        val scanner = Scanner(scannerConf)
+        val wfpString = buildString {
+            path.walk()
+                // TODO: Consider not applying the (somewhat arbitrary) blacklist.
+                .filter { !it.isDirectory && !BlacklistRules.hasBlacklistedExt(it.name) }
+                .forEach {
+                    ScanOss.log.info { "Computing fingerprint for file ${it.absolutePath}..." }
+                    val wfpString = createWfpForFile(it.path)
+                    append(wfpString)
```

Comment:
Maybe inline the `wfpString` variable here, esp. as the outer variable is called the same?